From 683eca02b43d0405422928e00f79049d054bfeb0 Mon Sep 17 00:00:00 2001
From: Alexander Bokovoy <abokovoy@redhat.com>
Date: Wed, 21 Nov 2012 18:33:49 +0200
Subject: [PATCH] trusts: replace use of python-crypto by m2crypto

python-crypto package is not available everywhere, use m2crypto instead.

Originally we thought to extend python-krbV to provide krb5_c_encrypt()
wrapper but m2crypto is readily available.

https://fedorahosted.org/freeipa/ticket/3271
---
 ipaserver/dcerpc.py | 6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ipaserver/dcerpc.py b/ipaserver/dcerpc.py
index 4eddbcdc304de9e50c5ccb9a71808fb71cf5b844..68b5ad52873753b6c78840c558636c16ea2dd803 100644
--- a/ipaserver/dcerpc.py
+++ b/ipaserver/dcerpc.py
@@ -41,7 +41,7 @@ from samba.ndr import ndr_pack
 from samba import net
 import samba
 import random
-from Crypto.Cipher import ARC4
+from M2Crypto import RC4
 try:
     from ldap.controls import RequestControl as LDAPControl #pylint: disable=F0401
 except ImportError:
@@ -526,8 +526,8 @@ class TrustDomainInstance(object):
 
     def generate_auth(self, trustdom_secret):
         def arcfour_encrypt(key, data):
-            c = ARC4.new(key)
-            return c.encrypt(data)
+            c = RC4.RC4(key)
+            return c.update(data)
         def string_to_array(what):
             blob = [0] * len(what)
 
-- 
1.7.11.7

